enhancements/local-storage/health-monitoring.md

```diff
@@ -0,0 +1,155 @@
+
+## Proposal
+
+[node_exporter](https://github.com/openshift/node_exporter) currently exports per-node metrics, we will use it to collect and export metrics for each device.
+
+### User Stories
+
+- As an admin, I would like to be alerted when a storage device is unhealthy and be linked to remediation steps.
+- As an admin, I would like if components (such as OpenShift Data Foundation) could access storage device health and failover faster.
+
+### Implementation Details/Notes/Constraints
+
+Implement a sidecar in the [node_exporter](https://github.com/openshift/node_exporter) component under [Cluster Monitoring Operator](https://github.com/openshift/cluster-monitoring-operator) that will scrape the metrics from the node and export them via the [textfile exporter](https://github.com/openshift/node_exporter/tree/master/text_collectors) (an exporter that scans `.prom` files in the
+`/var/node_exporter/textfile` directory and exports them.
+
+The Cluster Monitoring Operator will deploy the sidecar when the platform is Baremetal or None by modifying the `node_exporter` [daemonset](https://github.com/openshift/cluster-monitoring-operator/blob/master/assets/node-exporter/daemonset.yaml). The platform will be detected by querying the `infrastructure` resource.
+
+After that, the maintainance effort will mostly centered on the code the image runs to get health and update its `.prom` file. The logic for this collector can be in a subdirectory of CMO and have it's own OWNERS file allowing the ODF team to own it.
+
+We will be collecting metrics via [libstoragemgmt](https://github.com/libstorage/libstoragemgmt) and calling it via [libstoragemgmt-golang](https://github.com/libstorage/libstoragemgmt-golang/).
+If a block device is a partion or logical volume, we will only expose health for the parent device.
+
+
+### Ownership and Maintenance
+
+- The `node_exporter` daemonset creation is owned by the Openshift-Monitoring team. The PR for modifying the daemonset based on infrastructure to include the sidecar can be contributed by ODF, but Monitoring team will have to review.
+- The collector sidecar image and logic will live in a separate repository owned and maintained by the ODF team.
+- A new bugzilla subcomponent will be created under openshift-monitoring, but the default assignee will be ODF and ODF will be responsible for triaging the issues.
+
+### Risks and Mitigations
+
+- The golang bindings for libstoragemgmgt need some review and polish.
+  - Mitigation: The ODF(OpenShift Data Foundation) team will contribute to and help the RHEL team maintain libstoragemgmt golang
+
+## Design Details
+
+Each device will have a metric `storage_device_health` metric associated with it that has a value of 0,1, or 2. Where 0 is healthy state, 1 is a warning state, and 2 is a failed state. Each metric will be labeled by node, device-name, and device-id (where available).
```

Comment:
Seconded, it would be great to have some details regarding this. This could simply be smarts device health property or be based on various smart counters. How the later is calculated is an interesting detail.